src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonHighlightClickedBlocks.kt

```diff
@@ -1,56 +1,73 @@
 package at.hannibal2.skyhanni.features.dungeon
 
 import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.config.ConfigUpdaterMigrator
 import at.hannibal2.skyhanni.data.ClickType
 import at.hannibal2.skyhanni.events.BlockClickEvent
 import at.hannibal2.skyhanni.events.LorenzChatEvent
 import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
 import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
 import at.hannibal2.skyhanni.utils.BlockUtils
 import at.hannibal2.skyhanni.utils.BlockUtils.getBlockAt
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColor
+import at.hannibal2.skyhanni.utils.ExtendedChatColor
 import at.hannibal2.skyhanni.utils.LorenzColor
 import at.hannibal2.skyhanni.utils.LorenzVec
 import at.hannibal2.skyhanni.utils.RenderUtils.drawColor
 import at.hannibal2.skyhanni.utils.RenderUtils.drawString
+import at.hannibal2.skyhanni.utils.TimeLimitedCache
 import net.minecraft.init.Blocks
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.time.Duration.Companion.seconds
 
 @SkyHanniModule
 object DungeonHighlightClickedBlocks {
+    private val config get() = SkyHanniMod.feature.dungeon.clickedBlocks
+    private val blocks = TimeLimitedCache<LorenzVec, ClickedBlock>(3.seconds)
+    private var colourIndex = 0
+    private val colours = LorenzColor.entries.filter {
+        it !in listOf(
+            LorenzColor.BLACK,
+            LorenzColor.WHITE,
+            LorenzColor.CHROMA,
+            LorenzColor.GRAY,
+            LorenzColor.DARK_GRAY,
+        )
+    }
 
-    private val blocks = mutableListOf<ClickedBlock>()
-    private var colorIndex = 0
-    private val colors = listOf(LorenzColor.YELLOW, LorenzColor.AQUA, LorenzColor.GREEN, LorenzColor.LIGHT_PURPLE)
-
-    private fun getNextColor(): LorenzColor {
-        var id = colorIndex + 1
-        if (id == colors.size) id = 0
-        colorIndex = id
-        return colors[colorIndex]
+    private fun getRandomColour(): LorenzColor {
+        var id = colourIndex + 1
+        if (id == colours.size) id = 0
+        colourIndex = id
+        return colours[colourIndex]
     }
 
     @SubscribeEvent
     fun onChat(event: LorenzChatEvent) {
-        if (!SkyHanniMod.feature.dungeon.highlightClickedBlocks) return
-        if (!DungeonAPI.inDungeon()) return
+        if (!isEnabled()) return
 
         if (event.message == "§cYou hear the sound of something opening...") {
             event.blockedReason = "dungeon_highlight_clicked_block"
         }
+
+        if (event.message == "§cThat chest is locked!") {
+            blocks.lastOrNull { it.value.displayText.contains("Chest") }?.value?.colour = config.lockedChestColour.toChromaColor()
+        }
```

Comment:
These should prob get converted to repo patterns

Reply:
how would i do that?

Reply:
> how would i do that?

Shift + CTRL + F for RepoPattern.pattern in SkyHanni code base and look at any off them